builtin/game/register.lua

```diff
@@ -100,7 +101,19 @@ function core.register_entity(name, prototype)
 	name = check_modname_prefix(tostring(name))
 
 	prototype.name = name
-	prototype.__index = prototype  -- so that it can be used as a metatable
+
+	-- Backward compatibility to deprecated 'prototype.__index = prototype'
+	prototype.__index = function(entity, key)
+		local mod_origin = rawget(entity, "mod_origin")
+		if mod_origin and not (deprecated_index_warned and deprecated_index_warned[name]) then
+			core.log("warning", "Entity "..name.. " from mod "..mod_origin..
+					" uses deprecated setmetatable(item, builtin_item) method")
```

Comment:
```suggestion
					" uses deprecated setmetatable(item, {__index = ??) method")
```
Also hits on how to solve it would be helpful. Are they supposed to iterate through the table and copy all key/values to the destination table?

Reply:
@SmallJoker , the `setmetatable(item, builtin_item)` is wrong usage that should be deprecated. The right way / the correction proposal is `setmetatable(item, {__index = builtin_item})`.

Reply:
@SmallJoker, see https://github.com/minetest/minetest_game/pull/2328 for proposed solution in Minetest Game